packages/eslint-plugin/src/rules/prefer-nullish-coalescing.ts

```diff
@@ -279,6 +298,33 @@ export default util.createRule<Options, MessageIds>({
           return;
         }
 
+        if (tsutils.isUnionOrIntersectionType(type)) {
+          if (
+            ignorePrimitives!.string &&
+            type.types.some(t => tsutils.isTypeFlagSet(t, ts.TypeFlags.String))
+          ) {
+            return;
+          }
+          if (
+            ignorePrimitives!.boolean &&
+            type.types.some(t =>
+              tsutils.isTypeFlagSet(t, ts.TypeFlags.BooleanLiteral),
+            )
+          ) {
+            return;
+          }
```

Comment:
[Cleanup] These are handled by the `type.types.some` later on. Removing them doesn't cause any test failures. Unneeded code?

```suggestion
```

Reply:
Fixed both, thanks!